File: principalEmzima/pythonAutomatizacion/SaucedemoProjectPlaySync/features/pages/Description.py
from imports import Page  
from configconstan import BrowserConfig
from features.pages.Catalogue import Catalogue
from features.pages.Cart import Cart


class Description(Cart, Catalogue):
    def __init__(self, page: Page):
        """Inicializa el navegador en la instancia Description herendando de Catalogue que hereda de Login.
        Lo solicita para comenzar e incorporarlo en el resto de las funciones"""
        super().__init__(page)

    def verify_login(self):
        """Verifica si el usuario está autenticado antes de ejecutar otras funciones."""
        if not self.logged_in:
            raise RuntimeError("Debes iniciar sesión primero usando la función login() Heredada de Catalogue que la heredo de Login.")

    def add_to_cart_description(self):
        """Añade un producto al carro desde la página Description"""
        self.verify_login()
        self.page.locator('button[data-test="add-to-cart"][id="add-to-cart"]').click()
        return self

    def back_to_products(self):
        """Se devuelve a la página Catalogue"""
        self.verify_login()
        self.page.get_by_role('button',name='Back to products').click()
        return self

File: principalEmzima/jsAutomatizacion/SaucedemoProjectPlayCypress/cypress/pages/description.js
/// <reference types="cypress" />

const CatalogueClass = require('./catalogue');

class Description extends CatalogueClass {
    constructor() {
        super();
        this.backToProducts = () => cy.get('[data-test="back-to-products"]');
        this.addToCartDescription = () => cy.get('[data-test="add-to-cart"]');
        this.removeDescription = () => cy.get('[data-test="remove"]');
    }

    clickAddToCartDescription() {
        this.verifyLogin();
        this.addToCartDescription().should('exist').then(($addToCart) => {
            if ($addToCart.is(':visible')) {
                this.addToCartDescription().click();
            } else {
                cy.log('Ya existe el producto en el carro de compras')
            }
        });
    }

    clickRemoveDescription() {
        this.verifyLogin();
        this.removeDescription().should('exist').then(($remove) => {
            if ($remove.is(':visible')) {
                this.removeDescription.click();
            } else {
                cy.log('No se puede remover porque no existe el producto en el carro de compras')
            }
        });
    }

    clickBackToProducts() {
        this.verifyLogin();
        this.backToProducts().click();
    }
}


module.exports = Description;